apps/rehabstod/src/components/Form/SelectCareUnits.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { screen } from '@testing-library/react'
+import { beforeEach, expect, vi } from 'vitest'
+import { rest } from 'msw'
+import userEvent from '@testing-library/user-event'
+import { SelectCareUnits } from './SelectCareUnits'
+import { fakeUser, fakeUserPreferences } from '../../utils/fake'
+import { server } from '../../mocks/server'
+import { renderWithRouter } from '../../utils/renderWithRouter'
+import { User } from '../../schemas'
+
+const preferences = fakeUserPreferences()
+let onChange: (value: string) => void
+let user: User
+beforeEach(() => {
+  onChange = vi.fn()
+  user = fakeUser()
+})
+
+function getNumberOfUnitsInFakeUser(): number {
+  let numberOfUnits = 1
+  user.vardgivare.forEach((careProvider) => {
+    careProvider.vardenheter.forEach((careUnit) => {
+      numberOfUnits += 1
+      if (careUnit.mottagningar && careUnit.mottagningar.length > 0) {
+        careUnit.mottagningar.forEach(() => {
+          numberOfUnits += 1
+        })
+      }
+    })
+  })
+  return numberOfUnits
+}
```

Comment:
This could use the same `getUnits` function from the components with some tweaks if we move it out into a utils folder.